### ISSUE TITLE
Search - Select mode remains with empty state screen after deleting all expenses

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#46374]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.13-3
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team

## Action Performed:
Precondition:
- There is only one unpaid expense and no other expeneses in the account
1. Launch New Expensify app
2. Go to Search
3. Long press on the expense
4. Tap Select
5. Tap on the dropdown
6. Tap Delete
7. Delete the expense

## Expected Result:
The select mode will exit because there is no more expense

## Actual Result:
The select mode remains with empty state screen

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/user-attachments/assets/b46d228b-b8b7-41ec-b4e9-d410efdd630a

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by lanitochka17** [comment]
We think that this bug might be related to  #wave-collect - Release 1

**Comment by lanitochka17** [comment]
@sakluger  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by hungvu193** [comment]
Thanks for the proposals everyone! 

@nyomanjyotisa @cretadn22  I don't think updating a state inside a render statement is a good way to handle this issue.  


@dominictb 's [proposal here](https://github.com/Expensify/App/issues/46374#issuecomment-2254886628) looks good to me!

🎀 👀 🎀  C+ reviewed

**Comment by hungvu193** [comment]
Sorry for the notification, found a bug while testing selected proposal, I withdraw my [previous decision](https://github.com/Expensify/App/issues/46374#issuecomment-2257677625).

**Comment by hungvu193** [comment]
I was about to ask you for more information. Thanks, will test it shortly

> @hungvu193 I have updated the code changes in my alternative solution to make it easier for you to test. The general idea remains the same.

**Comment by hungvu193** [comment]
All good. Let's go with @dominictb 's [alternative proposal](https://github.com/Expensify/App/issues/46374#issuecomment-2254886628). cc @NikkiWines

**Comment by sakluger** [comment]
@hungvu193 - does @cretadn22's comment change anything?

**Comment by hungvu193** [comment]
> @hungvu193 - does @cretadn22's comment impact your recommendation?

No. I'll still keep my decision. Please assign @dominictb @NikkiWines

**Comment by NikkiWines** [comment]
Yep @dominictb's alt proposal looks good 👍

**Comment by hungvu193** [comment]
- [x] The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/44385
- [x] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: N/A
- [x] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: N/A
- [x] Determine if we should create a regression test for this bug: Yes.

This is a new feature that we added recently and still under development I believe so I don't think we need regression test for this one.

**Comment by sakluger** [comment]
Thanks @hungvu193. Could you please write out the proposed regression test steps?

**Comment by sakluger** [comment]
Summarizing payment on this issue:

Contributor: @dominictb $250, paid via Upwork
Contributor+: @hungvu193 $250, please request on Newdot

**Comment by hungvu193** [comment]
> Thanks @hungvu193. Could you please write out the proposed regression test steps?

My bad, I updated the [checklist](https://github.com/Expensify/App/issues/46374#issuecomment-2296099617)

**Comment by sakluger** [comment]
Ah, that makes more sense. Thanks!

**Comment by JmillsExpensify** [comment]
$250 approved for @hungvu193
